burrito_link/dllmain.c

```diff
@@ -396,3 +396,55 @@ uintptr_t mod_release() {
 extern __declspec(dllexport) void* get_release_addr() {
     return mod_release;
 }
+
+
+////////////////////////////////////////////////////////////////////////////////
+// Raidcore Nexus
+//
+// GetAddonDef()
+// 
+//
+//
+// These function is present to allow nexus to recognize this dll as a addon.
+// These function is the only function that is required all others are optional.
+////////////////////////////////////////////////////////////////////////////////
+struct AddonDefinition {
+	/* required */
+	signed int      Signature;      /* Raidcore Addon ID, set to random unqiue negative integer if not on Raidcore */
+	signed int      APIVersion;     /* Determines which AddonAPI struct revision the Loader will pass, use the NEXUS_API_VERSION define from Nexus.h */
+	const char*     Name;           /* Name of the addon as shown in the library */
+	struct AddonVersion {
+		signed short      Major;
+		signed short      Minor;
+		signed short      Build;
+		signed short      Revision;
+	} Version;
+	const char*     Author;         /* Author of the addon */
+	const char*     Description;    /* Short description */
+	void*           Load;           /* Pointer to Load Function of the addon */
+	void*           Unload;         /* Pointer to Unload Function of the addon. Not required if EAddonFlags::DisableHotloading is set. */
+	signed int      Flags;          /* Information about the addon */
+
+	/* update fallback */
+	signed int      Provider;       /* What platform is the the addon hosted on */
+	const char*     UpdateLink;     /* Link to the update resource */
+
+} AddonDef;
+
+extern __declspec(dllexport) struct AddonDefinition* GetAddonDef()
+{
+	AddonDef.Signature = -217;
```

Comment:
The comments in the type definition say to choose a random negative number, that's what I did.
```
signed int      Signature;      /* Raidcore Addon ID, set to random unqiue negative integer if not on Raidcore */
```
Even if it says the number must be unique, there were no problems when I used an already existing number.
The important part is to use a negative number, because official Raidcore-addons use positive numbers.

Reply:
I wonder what this number's use is then if it does not need to be unique. Maybe it is a feature delta has not yet added to nexus? Or "negative number" is just a catchall for "non raidcore plugin".

Reply:
I did some research and joined the developers discord (which is not as useful as I thought).
The signature is used to...
 - determine if the addon is from a third party
 - identify the addon so it does not get loaded more than once
 - manage dependencies with other addons (I could not find much information about this topic)

Maybe there are more features planned, I don't know.
All addons I checked use 6 random digits, should I change the value?

Reply:
Nice research

Looks like it is a regular signed integer, so we have 32 bits to play with. Lets  be cheeky here.

```
# Negative
1000 0000  0000 0000  0000 0000  0000 0000

# Burrito (ASCII)
B          r          t          o
0100 0010  0111 0010  0111 0100  0110 1111

# Negative & Burrito
1100 0010  0111 0010  0111 0100  0110 1111

# sint32
-1032686481
```

Lets go with
```
AddonDef.Signature = -1032686481;
```

Reply:
I like the way you think. :-)
Done.